homeassistant/components/netatmo/sensor.py

```diff
@@ -482,10 +482,11 @@ def update(self):
             self._state = None
             return
 
+        values = [x for x in data.values() if x is not None]
         if self._mode == 'avg':
-            self._state = round(sum(data.values()) / len(data), 1)
+            self._state = round(sum(values) / len(data), 1)
```

Comment:
We're dividing with the original data length here. Is that intended?

Reply:
The other option would be to replace `None` by 0 which didn’t make sense to me as it would change the result more drastically. Don’t you think?

Reply:
What does `None` represent when present as a value in the dict?

Generally the arithmetic mean should be calculated as the sum of the values divided by the number of values. Ie the number of observations should be the same in the dividend and the divisor.

Reply:
Oh, that is indeed an oversight by me. Thanks. I’ll fix that right away.